source/distributed/fullydistributed_tria.cc

```diff
@@ -0,0 +1,562 @@
+          }
+        return true;
+      }
+    } // namespace internal
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::reinit(
+      ConstructionData<dim, spacedim> &construction_data)
+    {
+      // check if there are locally relevant coarse-grid cells
+      if (construction_data.cells.size() == 0) // no
+        {
+          // 1) create a dummy hypercube
+          Point<spacedim> p1, p2;
+          for (unsigned int i = 0; i < dim; ++i)
+            {
+              p1(i) = 0.0;
+              p2(i) = 1.0;
+            }
+
+          std::vector<Point<spacedim>> vertices(
+            GeometryInfo<dim>::vertices_per_cell);
+          switch (dim)
+            {
+              case 1:
+                vertices[0] = p1;
+                vertices[1] = p2;
+                break;
+              case 2:
+                vertices[0] = vertices[1] = p1;
+                vertices[2] = vertices[3] = p2;
+
+                vertices[1](0) = p2(0);
+                vertices[2](0) = p1(0);
+                break;
+              case 3:
+                vertices[0] = vertices[1] = vertices[2] = vertices[3] = p1;
+                vertices[4] = vertices[5] = vertices[6] = vertices[7] = p2;
+
+                vertices[1](0) = p2(0);
+                vertices[2](1) = p2(1);
+                vertices[3](0) = p2(0);
+                vertices[3](1) = p2(1);
+
+                vertices[4](0) = p1(0);
+                vertices[4](1) = p1(1);
+                vertices[5](1) = p1(1);
+                vertices[6](0) = p1(0);
+
+
+                break;
+              default:
+                Assert(false, ExcNotImplemented());
+            }
+
+          std::vector<CellData<dim>> cells(1);
+          for (unsigned int i = 0; i < GeometryInfo<dim>::vertices_per_cell;
+               ++i)
+            cells[0].vertices[i] = i;
+          cells[0].material_id = 0;
+          SubCellData subcelldata;
+          dealii::parallel::Triangulation<dim, spacedim>::create_triangulation(
+            vertices, cells, subcelldata);
+
+          // 2) mark cell as artificial
+          for (auto cell = this->begin(); cell != this->end(); cell++)
+            {
+              cell->set_subdomain_id(dealii::numbers::artificial_subdomain_id);
+              cell->set_level_subdomain_id(
+                dealii::numbers::artificial_subdomain_id);
+            }
+
+          // 3) setup dummy mapping between locally relevant coarse-grid cells
+          //    and global cells
+          this->coarse_gid_to_lid.emplace_back(0, 0);
+          this->coarse_lid_to_gid.emplace_back(0, 0);
+        }
+      else
+        {
+          auto &boundary_ids = construction_data.boundary_ids;
+          auto &cells        = construction_data.cells;
+          auto &vertices     = construction_data.vertices;
+          auto &parts        = construction_data.parts;
+
+
+
+          // create inverse map
+          std::map<int, int> coarse_gid_to_lid;
```

Comment:
wrong types here!

Reply:
I know. This is older than `types::coarse_cell_id` and our discussions. I have started to update the data types yesterday and have not finished it yet. WIP ;)